Realm/RLMRealm.mm

```diff
@@ -581,13 +584,15 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration
         }
 
         DataInitializationFunction initializationFunction;
-        initializationFunction = [&isFirstOpen](SharedRealm) {
-            isFirstOpen = true;
-        };
+        if (!configuration.rerunOnOpen && configuration.initialSubscriptions) {
+            initializationFunction = [&isFirstOpen](SharedRealm) {
+                isFirstOpen = true;
+            };
+        }
 
         try {
             realm->_realm->update_schema(schema.objectStoreCopy, config.schema_version,
-                                         std::move(migrationFunction));
+                                         std::move(migrationFunction), std::move(initializationFunction));
```

Comment:
why do we need to add this, if the initialisationFunction is used?

Reply:
I don't understand the question. This is the only place that the initializationFunction is used.